Build failing with lt (lithuanian) language

Building mkdocs fails [after adding Lithuanian files](https://github.com/skynet-guide/skynet-guide.github.io/pull/42) and I don't know why.

For debugging I copied the working English file `docs/index.md` to ``docs/index.lt.md`` to rule out formatting errors, but I still get the same error, so I assume its a bug specifically with lt files.


Error message:
```
INFO    -  Cleaning site directory 
INFO    -  Building documentation to directory: /home/runner/work/skynet-guide.github.io/skynet-guide.github.io/site 
INFO    -  Building en documentation 
INFO    -  Building fr documentation 
INFO    -  Building de documentation 
INFO    -  Building ru documentation 
INFO    -  Building es documentation 
INFO    -  Building lt documentation 
ERROR   -  Error building page 'index.lt.md': partials/languages/lt.html 
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.7.10/x64/bin/mkdocs", line 8, in <module>
    sys.exit(cli())
  File "/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/click/core.py", line 1134, in __call__
    return self.main(*args, **kwargs)
  File "/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/click/core.py", line 1059, in main
    rv = self.invoke(ctx)
  File "/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/click/core.py", line 1665, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/click/core.py", line 1401, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/click/core.py", line 767, in invoke
    return __callback(*args, **kwargs)
  File "/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/mkdocs/__main__.py", line 152, in build_command
    build.build(config.load_config(**kwargs), dirty=not clean)
  File "/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/mkdocs/commands/build.py", line 295, in build
    config['plugins'].run_event('post_build', config=config)
  File "/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/mkdocs/plugins.py", line 96, in run_event
    result = method(**kwargs)
  File "/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/mkdocs_static_i18n/plugin.py", line 349, in on_post_build
    _build_page(file.page, config, files, nav, env, dirty)
  File "/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/mkdocs/commands/build.py", line 210, in _build_page
    output = template.render(context)
  File "/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/jinja2/environment.py", line 1289, in render
    self.environment.handle_exception()
  File "/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/jinja2/environment.py", line 924, in handle_exception
    raise rewrite_traceback_stack(source=source)
  File "/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/material/main.html", line 4, in top-level template code
    {% extends "base.html" %}
  File "/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/material/base.html", line 4, in top-level template code
    {% import "partials/language.html" as lang with context %}
  File "/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/jinja2/environment.py", line 1394, in make_module
    return TemplateModule(self, ctx)
  File "/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/jinja2/environment.py", line 1524, in __init__
    body_stream = list(template.root_render_func(context))  # type: ignore
  File "/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/material/partials/language.html", line 4, in top-level template code
    {% import "partials/languages/" ~ config.theme.language ~ ".html" as lang %}
  File "/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/jinja2/loaders.py", line 214, in get_source
    raise TemplateNotFound(template)
jinja2.exceptions.TemplateNotFound: partials/languages/lt.html
Error: Process completed with exit code 1.
```

--- a/mkdocs_static_i18n/plugin.py
+++ b/mkdocs_static_i18n/plugin.py
@@ -19,8 +19,14 @@ except ImportError:
 try:
     import pkg_resources
 
-    material_version = pkg_resources.get_distribution("mkdocs-material").version
+    material_dist = pkg_resources.get_distribution("mkdocs-material")
+    material_version = material_dist.version
+    material_languages = [
+        lang.split(".html")[0]
+        for lang in material_dist.resource_listdir("material/partials/languages")
+    ]
 except Exception:
+    material_languages = []
     material_version = None
 
 log = logging.getLogger("mkdocs.plugins." + __name__)
@@ -81,6 +87,9 @@ class I18nFiles(Files):
         expected_src_path = Path(path)
         expected_src_paths = [
             expected_src_path.with_suffix(f".{self.locale}{expected_src_path.suffix}"),
+            expected_src_path.with_suffix(
+                f".{self.default_locale}{expected_src_path.suffix}"
+            ),
             expected_src_path,
         ]
         return any(filter(lambda s: Path(s) in expected_src_paths, self.src_paths))
@@ -90,6 +99,9 @@ class I18nFiles(Files):
         expected_src_path = Path(path)
         expected_src_paths = [
             expected_src_path.with_suffix(f".{self.locale}{expected_src_path.suffix}"),
+            expected_src_path.with_suffix(
+                f".{self.default_locale}{expected_src_path.suffix}"
+            ),
             expected_src_path,
         ]
         for src_path in filter(lambda s: Path(s) in expected_src_paths, self.src_paths):
@@ -314,10 +326,12 @@ class I18n(BasePlugin):
         generate the navigation for the default site and per language.
         """
         main_files = I18nFiles([])
+        main_files.default_locale = self.default_language
         main_files.locale = self.default_language
         for language in self.all_languages:
             self.i18n_configs[language] = deepcopy(config)
             self.i18n_files[language] = I18nFiles([])
+            self.i18n_files[language].default_locale = self.default_language
             self.i18n_files[language].locale = language
 
         base_paths = set()
@@ -427,7 +441,12 @@ class I18n(BasePlugin):
 
             # Support mkdocs-material i18n search context
             if config["theme"].name == "material":
-                config["theme"].language = language
+                if language in material_languages:
+                    config["theme"].language = language
+                else:
+                    log.warning(
+                        f"Language {language} is not supported by mkdocs-material=={material_version}, setting 'theme.language: {self.default_language}'"
+                    )
 
             # Run `nav` plugin events.
             # This is useful to be compatible with nav order changing plugins

--- a/mkdocs_static_i18n/plugin.py
+++ b/mkdocs_static_i18n/plugin.py
@@ -445,7 +445,9 @@ class I18n(BasePlugin):
                     config["theme"].language = language
                 else:
                     log.warning(
-                        f"Language {language} is not supported by mkdocs-material=={material_version}, setting 'theme.language: {self.default_language}'"
+                        f"Language {language} is not supported by "
+                        f"mkdocs-material=={material_version}, not setting "
+                        "the 'theme.language' option"
                     )
 
             # Run `nav` plugin events.

--- a/tests/test_structure.py
+++ b/tests/test_structure.py
@@ -31,7 +31,9 @@ def test_build_use_directory_urls(config_base):
     site_dir = config_base["site_dir"]
     build(config_base)
     generate_site = [f.relative_to(site_dir) for f in Path(site_dir).glob("**/*.html")]
-    generate_site.extend([f.relative_to(site_dir) for f in Path(site_dir).glob("**/image*.png")])
+    generate_site.extend(
+        [f.relative_to(site_dir) for f in Path(site_dir).glob("**/image*.png")]
+    )
     print(list(Path(site_dir).glob("**/*.html")))
     assert sorted(generate_site) == sorted(USE_DIRECTORY_URLS)
 
@@ -41,7 +43,9 @@ def test_build_no_use_directory_urls(config_base):
     site_dir = config_base["site_dir"]
     build(config_base)
     generate_site = [f.relative_to(site_dir) for f in Path(site_dir).glob("**/*.html")]
-    generate_site.extend([f.relative_to(site_dir) for f in Path(site_dir).glob("**/image*.png")])
+    generate_site.extend(
+        [f.relative_to(site_dir) for f in Path(site_dir).glob("**/image*.png")]
+    )
     print(list(Path(site_dir).glob("**/*.html")))
     assert sorted(generate_site) == sorted(NO_USE_DIRECTORY_URLS)
 
@@ -83,7 +87,9 @@ def test_plugin_use_directory_urls(config_plugin):
     site_dir = config_plugin["site_dir"]
     build(config_plugin)
     generate_site = [f.relative_to(site_dir) for f in Path(site_dir).glob("**/*.html")]
-    generate_site.extend([f.relative_to(site_dir) for f in Path(site_dir).glob("**/image*.png")])
+    generate_site.extend(
+        [f.relative_to(site_dir) for f in Path(site_dir).glob("**/image*.png")]
+    )
     print(list(Path(site_dir).glob("**/*.html")))
     assert sorted(generate_site) == sorted(PLUGIN_USE_DIRECTORY_URLS)
 
@@ -93,7 +99,9 @@ def test_plugin_no_use_directory_urls(config_plugin):
     site_dir = config_plugin["site_dir"]
     build(config_plugin)
     generate_site = [f.relative_to(site_dir) for f in Path(site_dir).glob("**/*.html")]
-    generate_site.extend([f.relative_to(site_dir) for f in Path(site_dir).glob("**/image*.png")])
+    generate_site.extend(
+        [f.relative_to(site_dir) for f in Path(site_dir).glob("**/image*.png")]
+    )
     print(list(Path(site_dir).glob("**/*.html")))
     assert sorted(generate_site) == sorted(PLUGIN_NO_USE_DIRECTORY_URLS)
 
@@ -129,7 +137,9 @@ def test_plugin_use_directory_urls_no_default_language(
     site_dir = config_plugin_no_default_language["site_dir"]
     build(config_plugin_no_default_language)
     generate_site = [f.relative_to(site_dir) for f in Path(site_dir).glob("**/*.html")]
-    generate_site.extend([f.relative_to(site_dir) for f in Path(site_dir).glob("**/image*.png")])
+    generate_site.extend(
+        [f.relative_to(site_dir) for f in Path(site_dir).glob("**/image*.png")]
+    )
     print(list(Path(site_dir).glob("**/*.html")))
     assert sorted(generate_site) == sorted(PLUGIN_USE_DIRECTORY_URLS_NO_DEFAULT)
 
@@ -141,6 +151,8 @@ def test_plugin_no_use_directory_urls_no_default_language(
     site_dir = config_plugin_no_default_language["site_dir"]
     build(config_plugin_no_default_language)
     generate_site = [f.relative_to(site_dir) for f in Path(site_dir).glob("**/*.html")]
-    generate_site.extend([f.relative_to(site_dir) for f in Path(site_dir).glob("**/image*.png")])
+    generate_site.extend(
+        [f.relative_to(site_dir) for f in Path(site_dir).glob("**/image*.png")]
+    )
     print(list(Path(site_dir).glob("**/*.html")))
     assert sorted(generate_site) == sorted(PLUGIN_NO_USE_DIRECTORY_URLS_NO_DEFAULT)

